### PR TITLE
Add xi.itemFlag enum

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [N/A] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

Add xi.itemFlag enum

## Steps to test these changes

N/A

### CODE REVIEW

File: src/map/lua/lua_item.cpp [hunk 1]
```diff
@@ -149,6 +149,16 @@ bool CLuaItem::isSubType(uint8 subtype)
     return m_PLuaItem->isSubType(static_cast<ITEM_SUBTYPE>(subtype));
 }
 
+bool CLuaItem::isRare()
+{
+    return m_PLuaItem->getFlag() & ITEM_FLAG_RARE;
```
Comment (TeoTwawki) [inline src/map/lua/lua_item.cpp]: suggestion: collapse these to one function and pass the enum or values in as input so one can check either flag or even multiple as mask bits.

See canUseMisc for an example of what I mean.

Comment (zach2good) [reply src/map/lua/lua_item.cpp]: +1 for this, will change when I have time

**Comment by zach2good** [comment]
I don't think bindings for something as simple as checking flags is really needed anymore, provided we have more human readable enums:

```lua
local isEx = bit.band(item:getFlag(), xi.itemFlag.EX) ~= 0
```

**Review by WinterSolstice8 — APPROVED** [review]
+1 for built in tutorial

**Review by claywar — APPROVED** [review]

**Pull request merged**